web/src/app/services/selection.service.ts

```diff
@@ -55,19 +55,22 @@ export class SelectionService {
   }
 
   public toggleCourse(course: Listing) {
-    
+    let closedCount = -1;
     if (this.hasSelectedSection(course)) {
       let store = this.getSelections();
       delete store[course.id];
       this.setItem('selections', JSON.stringify(store));
     } else {
+      closedCount = 0;
       course.sections.forEach((s) => {
-        if (s.seatsTaken < s.seats) {
-          this.addSection(s);
+        if (s.seatsTaken >= s.seats) {
```

Comment:
I don't think this logic should go here. The number of closed sections isn't really related to toggling a course, so I think that should go somewhere else, probably the course component